src/legacy/core_plugins/kibana/public/discover/plugin.ts

```diff
@@ -33,28 +41,108 @@ import {
+      order: -1004,
+      euiIconType: 'discoverApp',
+      mount: async (context, params) => {
+        if (!this.initializeServices) {
+          throw Error('Discover plugin method initializeServices is undefined');
+        }
+        if (!this.initializeInnerAngular) {
+          throw Error('Discover plugin method initializeInnerAngular is undefined');
+        }
+        await this.initializeServices();
+        await this.initializeInnerAngular();
+        const { renderApp } = await import('./application');
+        return renderApp(innerAngularName, params.element);
+      },
+    });
   }
 
   start(core: CoreStart, plugins: DiscoverStartPlugins): DiscoverStart {
-    // TODO enable this when possible, seems it broke a functional test:
-    // dashboard mode Dashboard View Mode Dashboard viewer can paginate on a saved search
-    // const factory = new SearchEmbeddableFactory(plugins.uiActions.executeTriggerActions);
-    // plugins.embeddable.registerEmbeddableFactory(factory.type, factory);
+    // we need to register the application service at setup, but to render it
+    // there are some start dependencies necessary, for this reason
+    // initializeInnerAngular + initializeServices are assigned at start and used
+    // when the application/embeddable is mounted
+    this.initializeInnerAngular = async () => {
+      if (this.innerAngularInitialized) {
+        return;
+      }
+      // this is used by application mount and tests
+      const module = getInnerAngularModule(innerAngularName, core, plugins);
+      setAngularModule(module);
+      this.innerAngularInitialized = true;
+    };
+
+    this.initializeServices = async (test = false) => {
+      if (this.servicesInitialized) {
+        return;
+      }
+      const services = await buildServices(core, plugins, test);
+      setServices(services);
+      this.servicesInitialized = true;
+    };
+
+    this.registerEmbeddable(core, plugins);
+    registerFeature();
   }
 
-  stop() {}
+  /**
+   * register embeddable with a slimmer embeddable version of inner angular
+   */
+  private async registerEmbeddable(core: CoreStart, plugins: DiscoverStartPlugins) {
+    const { SearchEmbeddableFactory } = await import('./embeddable');
+    const getInjector = async () => {
+      if (!this.initializeServices) {
+        throw Error('Discover plugin registerEmbeddable:  initializeServices is undefined');
+      }
+      await this.initializeServices();
+      getInnerAngularModuleEmbeddable(embeddableAngularName, core, plugins);
+      const mountpoint = document.createElement('div');
+      return angular.bootstrap(mountpoint, [embeddableAngularName]);
+    };
+    const isEditable = () => core.application.capabilities.discover.save as boolean;
+
+    const factory = new SearchEmbeddableFactory(
+      plugins.uiActions.executeTriggerActions,
+      getInjector,
+      isEditable
+    );
+    plugins.embeddable.registerEmbeddableFactory(factory.type, factory);
```

Comment:
You may want to load this plugin class via a `hacks` uiExport so that this embeddable is always registered regardless of which app is on the page.

Reply:
This is done this way, else it wouldn't work